### PR TITLE
Tracecontext extension

### PR DESCRIPTION
**Pull request by dstenroejl**
## Description

* Clean up in comments
* Add FunctionApp tests that set `traceparent` header
* Rename `TraceContext` to `TraceParent`
* Added Function App extension `AddApplicationInsights`

## References

https://app.zenhub.com/workspaces/mighty-ducks---the-outlaws-6193fe815d79fc0011e741b1/issues/energinet-datahub/the-outlaws/346

### CODE REVIEW

**Review by lasrinnil — DISMISSED** [review]
var traceParent = TraceParent.Parse(context.TraceContext.TraceParent);

This makes it more clear what is exposed :) GJ!

File: source/App/source/Common.Abstractions/Common.Abstractions.csproj [hunk 1]
```diff
@@ -55,11 +55,11 @@ limitations under the License.
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.SourceLink.GitHub" Version="1.0.0">
+    <PackageReference Include="Microsoft.SourceLink.GitHub" Version="1.1.1">
       <PrivateAssets>all</PrivateAssets>
       <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
     </PackageReference>
-    <PackageReference Include="NodaTime" Version="3.1.0" />
+    <PackageReference Include="NodaTime" Version="3.1.2" />
```
Comment (lasrinnil) [inline source/App/source/Common.Abstractions/Common.Abstractions.csproj:62]: Is this intended? Perhaps the release note should mention that the package is updated?

File: source/App/source/FunctionApp/Extensions/DependencyInjection/ApplicationInsightsExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+
+namespace Energinet.DataHub.Core.App.FunctionApp.Extensions.DependencyInjection
+{
+    /// <summary>
+    /// Extensions to <see cref="Microsoft.Extensions.DependencyInjection.IServiceCollection"/>
+    /// that allow adding Application Insights services to Function App's.
+    /// </summary>
+    public static class ApplicationInsightsExtensions
```
Comment (lasrinnil) [inline source/App/source/FunctionApp/Extensions/DependencyInjection/ApplicationInsightsExtensions.cs:25]: This class extends ServiceCollection and not ApplicationInsight.

Consider naming it `ServiceCollectionExtensions`

**Review by lasrinnil — APPROVED** [review]

File: source/App/source/FunctionApp/Extensions/DependencyInjection/ApplicationInsightsExtensions.cs (hunk 2 above)
Comment (dstenroejl) [reply source/App/source/FunctionApp/Extensions/DependencyInjection/ApplicationInsightsExtensions.cs:25]: Good observation, it is on purpose. I have looked at similar extensions from Microsoft and they use names like this. I think it makes sense to subcategorize extensions in the "DependencyInjection" namespace.

File: source/App/source/Common.Abstractions/Common.Abstractions.csproj (hunk 1 above)
Comment (dstenroejl) [reply source/App/source/Common.Abstractions/Common.Abstractions.csproj:62]: Will do

**Review by lasrinnil — APPROVED** [review]

**Pull request merged**